### PR TITLE
Update units-of-measure.md

### PR DESCRIPTION
**Pull request by smoothdeveloper**
Add a mention that F# can support the feature for other types, for type safety purpose (not only arithmetic), and the fact it is zero cost abstraction.

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [docs/fsharp/language-reference/units-of-measure.md](https://github.com/dotnet/docs/blob/283d7ea2dd72f1d82e9ecc3bbdaf5310d4aa4585/docs/fsharp/language-reference/units-of-measure.md) | [Units of Measure](https://review.learn.microsoft.com/en-us/dotnet/fsharp/language-reference/units-of-measure?branch=pr-en-us-37468) |

### CODE REVIEW

File: docs/fsharp/language-reference/units-of-measure.md [hunk 1]
```diff
@@ -7,6 +7,8 @@ ms.date: 08/15/2020
 
 Floating point and signed integer values in F# can have associated units of measure, which are typically used to indicate length, volume, mass, and so on. By using quantities with units, you enable the compiler to verify that arithmetic relationships have the correct units, which helps prevent programming errors.
 
+> [!NOTE] despite the units of measure main raison d'être is correctness in arithmetic computations, the feature can also be leveraged for adding type safe annotation with zero representation costs to other types, with approach such as [FSharp.UMX](https://github.com/fsprojects/FSharp.UMX) project.
```
Comment (BillWagner) [inline docs/fsharp/language-reference/units-of-measure.md:10]: Making a few comments here:

First, the NOTE tag needs to be on its own line for the build system to render it correctly.
Second, we avoid non-English phrases in our docs. Among other issues, it wreaks havoc on localization to other languages.

I made some suggestions. Feel free to accept them, or offer your own alternative wording:

```suggestion
> [!NOTE]
> These examples demonstrates correctness in arithmetic computations involving units of measure, the feature can also be leveraged for adding type safe annotation with zero representation costs to other types, with approach such as [FSharp.UMX](https://github.com/fsprojects/FSharp.UMX) project.
```

**Review by BillWagner — CHANGES_REQUESTED** [review]
Hi @smoothdeveloper 

Thanks for adding this clarification. I've made a suggestion based on a couple writing style rules we have.

Once you accept them, or make other updates, I can merge this.

File: docs/fsharp/language-reference/units-of-measure.md (hunk 1 above)
Comment (smoothdeveloper) [reply docs/fsharp/language-reference/units-of-measure.md:10]: @BillWagner thanks a bunch for blending my submission to make it fit both the writing style and display adjustments required to make it fit nicely in the F# documentation.

The adjustments looks great to me.

**Review by BillWagner — APPROVED** [review]

**Comment by BillWagner** [comment]
Thanks again @smoothdeveloper 

I've merged this.

**Review by vzarytovskii — APPROVED** [review]

**Pull request merged**